acceptance/src/data.rs

```diff
@@ -86,7 +112,7 @@ pub async fn assert_scan_data(engine: Arc<dyn Engine>, test_case: &TestCaseInfo)
     let table_root = test_case.table_root()?;
     let table = Table::new(table_root);
     let snapshot = table.snapshot(engine, None)?;
-    let scan = ScanBuilder::new(snapshot).build();
+    let scan = ScanBuilder::new(snapshot).build()?;
```

Comment:
not really related to this PR, but I remember wondering if it would be more ergonomic, if the `Snapshot` could be transformed into a `ScanBuilder`, may via something like `snapshot.build_scan()`?

Reply:
that's not a bad idea, but I think we'd still want something like `snapshot.scan_builder()` because we can set stuff on the scan via the builder and we don't want to do it all via args, so you'd want to be able to do something like:
```rust
let scan = snapshot().scan_builder().with_schema(schema).predicate(pred).build()?;
```

Reply:
Added #206 for this since it would be a nice first issue